repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/regexrbac/RegexRbacAccessTest.groovy

```diff
@@ -0,0 +1,228 @@
+
+    def setupSpec() {
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort)
+
+        def params = properties.getDefaultTemplateParams()
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/regexrbac/common", params)
+        repose.start()
+        repose.waitForNon500FromUrl(reposeEndpoint)
+    }
+
+    @Unroll
+    def "a request for #resource which matches a defined resource should succeed"() {
+        when: "a request is made to Repose"
+        MessageChain messageChain = deproxy.makeRequest(
+            url: reposeEndpoint + resource,
+            method: GET,
+            headers: [(ROLES_HEADER): "any"]
+        )
+
+        then: "the origin service responds with a 200"
+        messageChain.handlings.size() == 1
+        messageChain.receivedResponse.code as Integer == SC_OK
+
+        where:
+        resource << ["/url/caSing", "/url/regex/1234", "/url/regex/9999", "/url/regex/foo"]
+    }
+
+    @Unroll
+    def "a request for #resource which does not match any defined resource should be rejected"() {
+        when: "a request is made to Repose"
+        MessageChain messageChain = deproxy.makeRequest(
+            url: reposeEndpoint + resource,
+            method: GET,
+            headers: [(ROLES_HEADER): "any"]
+        )
+
+        then: "Repose responds with a 404"
+        messageChain.handlings.size() == 0
+        messageChain.receivedResponse.code as Integer == SC_NOT_FOUND
+
+        where:
+        resource << ["/url/casing", "/url/regex/12345", "/not/a/path"]
+    }
+
+    @Unroll
+    def "a request with #roles for #resource should succeed"() {
+        when: "a request is made to Repose"
+        MessageChain messageChain = deproxy.makeRequest(
+            url: reposeEndpoint + resource,
+            method: GET,
+            headers: roles ? [(ROLES_HEADER): roles] : []
+        )
+
+        then: "the origin service responds with a 200"
+        messageChain.handlings.size() == 1
+        messageChain.receivedResponse.code as Integer == SC_OK
+
+        where:
+        resource                    | roles
+        "/all"                      | null
+        "/any"                      | null
+        "/all"                      | "role1"
+        "/all"                      | "9elor"
+        "/any"                      | "role1"
+        "/any"                      | "9elor"
+        "/all"                      | "role1, role2"
+        "/any"                      | "role1, role2"
+        "/all/any"                  | "role1"
+        "/any/all"                  | "role1"
+        "/roles/simple"             | "role1"
+        "/roles/simple"             | "role2"
+        "/roles/simple"             | "role1,role2"
+        "/roles/simple"             | "role1, role2"
+        "/roles/nbsp"               | "nb sp"
+        "/roles/nbsp/end"           | "nbsp "
```

Comment:
This test currently fails for me locally. The reason is that the space is at the end of the role. The configured role of `nbsp ` has the `&nbsp;` character translated into a normal space _after_ trimming occurs. The `HttpServletRequestWrapper`, however, trims the space around all header values. The result is that the translated, configured `nbsp ` role does not match the `nbsp` role pulled from the request.

So we can either:

1. Trim the space around configured roles _after_ replacing the `&nbsp;` character. We would just have to be clear that leading/trailing whitespace, including `&nbsp;` is ignored. Also, I'm not sure if a role can exist that starts or ends with a space in Identity. If it can, then we do not have a way to account for that at the moment.
1. Consider this test invalid and remove it.

Reply:
Option was chosen, so this test should pass now.